cmd/soroban-cli/src/config/network.rs

```diff
@@ -97,6 +116,17 @@ pub struct Network {
         help_heading = HEADING_RPC,
     )]
     pub rpc_url: String,
+    /// Optional header (e.g. API Key) to include in requests to the RPC
+    #[arg(
+        long = "rpc-header",
+        env = "STELLAR_RPC_HEADER",
+        help_heading = HEADING_RPC,
+        num_args = 1,
+        action = clap::ArgAction::Append,
+        value_delimiter = ',',
```

Comment:
For more general purposes, we may need to consider whether we should use `,` as a separator, because , is likely to appear in the value. For example, my User-Agent contains it: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36`

Reply:
I like the idea of using new line as you did in the other example, given that new lines are the natural delimiter. New lines may create issues in some environments for env vars, but also, we expect only a single header to be used so I'm not sure it's worth dwelling on that.